x/evm/keeper/grpc_query.go

```
@@ -754,3 +755,197 @@ func getChainID(ctx sdk.Context, chainID int64) (*big.Int, error) {
+		panic("context invalid")
+	}
+
+	codeHash := common.HexToHash(req.CodeHash)
+
+	code := k.GetCode(*ctx, codeHash)
+	return &types.GetCodeResponse{
+		Code: code,
+	}, nil
+}
+
+// PostSetAccountStateDB sets account in statedb for sgx
+func (k Keeper) PostSetAccountStateDB(_ context.Context, req *types.SetAccountRequest) (*types.SetAccountResponse, error) {
+	ctx := k.sdkCtxs[req.HandlerId]
+	if ctx == nil {
+		panic("context invalid")
+	}
+
+	addr := common.HexToAddress(req.Addr)
+
+	var account statedb.Account
+	err := json.Unmarshal(req.Account, &account)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	err = k.SetAccount(*ctx, addr, account)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	return &types.SetAccountResponse{Nonce: account.Nonce, CodeHash: account.CodeHash}, nil
+}
+
+// PostSetStateStateDB sets state in statedb for sgx
+func (k Keeper) PostSetStateStateDB(_ context.Context, req *types.SetStateRequest) (*types.SetStateResponse, error) {
+	ctx := k.sdkCtxs[req.HandlerId]
+	if ctx == nil {
+		panic("context invalid")
+	}
+
+	addr := common.HexToAddress(req.Addr)
+	key := common.HexToHash(req.Key)
+
+	k.SetState(*ctx, addr, key, req.Value)
+	return &types.SetStateResponse{}, nil
+}
+
+// PostSetCodeStateDB sets code in statedb for sgx
+func (k Keeper) PostSetCodeStateDB(_ context.Context, req *types.SetCodeRequest) (*types.SetCodeResponse, error) {
+	ctx := k.sdkCtxs[req.HandlerId]
+	if ctx == nil {
+		panic("context invalid")
+	}
+
+	k.SetCode(*ctx, req.CodeHash, req.Code)
+	return &types.SetCodeResponse{}, nil
+}
+
+// PostDeleteAccountStateDB delete account in statedb for sgx
+func (k Keeper) PostDeleteAccountStateDB(_ context.Context, req *types.DeleteAccountRequest) (*types.DeleteAccountResponse, error) {
+	ctx := k.sdkCtxs[req.HandlerId]
+	if ctx == nil {
+		panic("context invalid")
+	}
+
+	addr := common.HexToAddress(req.Addr)
+
+	k.DeleteAccount(*ctx, addr)
```

let's return an error if there is

---

@amaury1093 done.